scripts/usb_scan.py

```diff
@@ -0,0 +1,545 @@
+#!/usr/bin/env python
+#
+# Copyright (C) Citrix Systems Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation; version 2.1 only. #
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# called by XAPI to do
+# 1. a "lsusb" operation: get all usb device and interface information
+# 2. check if device can pass through based on policy file
+# 3. return the device to pass through in json format
+#
+
+import abc
+import argparse
+import json
+import xcp.logger as log
+import logging
+import pyudev
+import re
+import traceback
+
+# True to print log to user.log
+__DIAGNOSTIC = False
+
+
+class UsbObject(dict):
+    __metaclass__ = abc.ABCMeta
```

Comment:
Please add doc strings for all classes and functions in them

Reply:
TODO